presto-main/src/main/java/com/facebook/presto/server/HttpRemoteTask.java

```diff
@@ -301,17 +310,22 @@ public synchronized void setOutputBuffers(OutputBuffers newOutputBuffers)
     }
 
     @Override
-    public int getPartitionedSplitCount()
+    public synchronized int getPartitionedSplitCount()
+    {
+        return pendingSourceSplitCount + taskInfo.get().getStats().getQueuedPartitionedDrivers() + taskInfo.get().getStats().getRunningPartitionedDrivers();
+    }
+
+    @Override
+    public synchronized int getQueuedPartitionedSplitCount()
     {
-        int splitCount = pendingSourceSplitCount;
-        return splitCount + taskInfo.get().getStats().getQueuedPartitionedDrivers() + taskInfo.get().getStats().getRunningPartitionedDrivers();
+        return pendingSourceSplitCount + taskInfo.get().getStats().getQueuedPartitionedDrivers();
     }
 
     @Override
-    public int getQueuedPartitionedSplitCount()
+    public synchronized void removeSplitCountChangeListener()
```

Comment:
I don't think you need this method. If `updateTaskInfo` is called when it's already in a "done" state it ignores the update, and splits can't be added if it's in a done state either, or did I miss something?